exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -287,3 +297,35 @@ func encodingToComponentID(encoding string) (*component.ID, error) {
 	id := component.NewID(componentType)
 	return &id, nil
 }
+
+func messagesWithHeaders(msg []*sarama.ProducerMessage, h []sarama.RecordHeader) {
+	if len(h) == 0 || len(msg) == 0 {
+		return
+	}
+	for i := range msg {
+		if len(msg[i].Headers) == 0 {
+			msg[i].Headers = h
+			continue
+		}
+		msg[i].Headers = append(msg[i].Headers, h...)
+	}
+}
+
+func metadataToHeaders(ctx context.Context, keys []string) []sarama.RecordHeader {
+	if len(keys) == 0 {
+		return nil
+	}
+	info := client.FromContext(ctx)
+	headers := make([]sarama.RecordHeader, 0, len(keys))
+	for _, key := range keys {
+		v := info.Metadata.Get(key)
+		if len(v) != 1 {
+			continue
+		}
+		headers = append(headers, sarama.RecordHeader{
+			Key:   []byte(key),
+			Value: []byte(v[0]),
+		})
```

Comment:
Could we handle multiple values by creating a RecordHeader for each value, with the same key?